src/odd.rs

```diff
@@ -153,6 +161,40 @@ impl Odd<BoxedUint> {
     }
 }
 
+#[cfg(feature = "serde")]
+impl<'de, T: Deserialize<'de> + Integer + Zero> Deserialize<'de> for Odd<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: T = T::deserialize(deserializer)?;
+
+        if bool::from(value.is_odd()) {
+            Ok(Self(value))
+        } else if bool::from(value.is_zero()) {
+            Err(D::Error::invalid_value(
+                Unexpected::Other("zero"),
+                &"a non-zero odd value",
+            ))
+        } else {
+            Err(D::Error::invalid_value(
+                Unexpected::Other("even"),
+                &"an odd value",
+            ))
+        }
```

Comment:
Seems like this is largely duplicating `Odd::new`. And why are you checking for zero explicitly here? Zero is even and will fail that way.

Reply:
`Odd::new`? I copied the `NonZero` deserialization code almost verbatim!

Reply:
Mathematically it seems weird to special case zero here, IMO.

I guess you can claim it's more helpful, but when is it actually helpful to know the value is zero beyond simply being even?

Reply:
> I copied the NonZero deserialization code almost verbatim!

Well, that code actually has a reason to worry about zero as a value specifically, and it would probably be good if that were updated to use `NonZero::new` as well.

Reply:
I removed the superfluous error message. Not sure what you mean by `Odd::new` though. Are you saying I should change this PR to use the `pub fn new(n: T) -> CtOption<Self>` and map the case the `CtOption` is `None` to a Serde error?

Reply:
It's six of one, half dozen of another, but yes, that would avoid repeating the logic to validate `Odd`'s invariant